nextmv/cloud/application.py

```diff
@@ -717,13 +719,13 @@ def _run_result(
             endpoint=f"{self.endpoint}/runs/{run_id}",
             query_params=query_params,
         )
+        result = RunResult.from_dict(response.json())
         if not large_output:
-            return RunResult.from_dict(response.json())
+            return result
 
-        download_url = DownloadURL.from_dict(response.json())
-        response2 = self.client.request(
-            method="GET",
-            endpoint=download_url.url,
-        )
+        download_url = DownloadURL.from_dict(response.json()["output"])
+        download_response = requests.get(download_url.url)
+        download_response.raise_for_status()
+        result.output = download_response.json()
```

Comment:
```suggestion
        download_response = self.client.request(
            method="GET",
            endpoint=download_url.url,
            headers={"Content-Type": "application/json"},
        )
        result.output = download_response.json()
```

Actually I think this is what we should do. I introduced the `headers` argument in the `request` method in case one doesn’t want to use the client’s default headers.

Reply:
Yeah, good call. That is clean again. :relaxed: